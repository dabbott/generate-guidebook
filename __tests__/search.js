const { Volume, createFsFromVolume } = require('memfs')
const scan = require('..')
const { buildIndex, exportIndex } = require('../search')

function createFs(fileTree) {
  const volume = Volume.fromNestedJSON(fileTree, '/')
  const memfs = createFsFromVolume(volume)
  return memfs
}

describe('search', () => {
  it('builds a search index', () => {
    const fs = createFs({
      pages: {
        'index.mdx': 'hello',
        'a.mdx': 'foo',
        a: {
          '1.mdx': 'foo',
        },
      },
    })

    const root = scan('/pages', fs)
    const index = buildIndex('/pages', root, fs)

    expect(index.search('hello')).toEqual([0])
    expect(index.search('fo')).toEqual([1, 2])

    expect(exportIndex(index)).toMatchSnapshot()
  })

  it('indexes titles', () => {
    const fs = createFs({
      pages: {
        'index.mdx': '# hello',
      },
    })

    const root = scan('/pages', fs)
    const index = buildIndex('/pages', root, fs)

    expect(index.search('hello')).toEqual([0])
  })
})
