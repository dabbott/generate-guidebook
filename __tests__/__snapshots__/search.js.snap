// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search builds a search index 1`] = `"[[{\\"h\\":[0],\\"he\\":[0],\\"hel\\":[0],\\"hell\\":[0],\\"hello\\":[0],\\"f\\":[1,2],\\"fo\\":[1,2],\\"foo\\":[1,2]},{},{},{},{},{},{},{},{}],{},[\\"@0\\",\\"@1\\",\\"@2\\"]]"`;
